### ISSUE TITLE
[css-fonts] General function of system font families for parity with corresponding generic

### ISSUE THREAD

**Issue by bleper** [w3c/csswg-drafts#10852]
System font families as they exist are not required to be mapped to anything. This has the advantage that implementations do not have to make any effort to conform to the specification, but the disadvantage that they have neither the incentive to change things nor the benefit of supporting null mapping

This impacts authors by requiring duplication for a straightforward use case, while fragmenting the platform further, leaving more inconsistency between browser vendors, increasing cognitive costs in terms of learning and documentation

A solution may be a function that takes the name of a generic font family, outputting the corresponding system font, otherwise the generic one. At this time, for compatibility, the current system font families could be aliases for the output of such a function in cases which now remain unsupported, and non-overlapping areas before their elimination should be compensated at this level of abstraction, leaving more room for future expansion of system font support or introduction of new types of generic families

For example, such function could be called `ui()` and for the input `font-family: ui(rounded)` it would exist as the equivalent of `font-family: ui-rounded, ui-sans-serif, system-ui, rounded, sans-serif`

Ultimately, the pain point will be solved by having the feature necessarily match some fonts, leaving the exact behavior to the vendor's choice

**Comment by svgeesus** [comment]
> System font families as they exist are not required to be mapped to anything.

That is incorrect. From [10.4. System Font](https://drafts.csswg.org/css-fonts-4/#system-font):

> The System Font is the font which is used by the [system-ui](https://drafts.csswg.org/css-fonts-4/#valdef-font-family-system-ui) [generic font family name](https://drafts.csswg.org/css-fonts-4/#system-ui-def). It is an example of a Preinstalled Font.

following that link takes us to [`system-ui`](https://drafts.csswg.org/css-fonts-4/#valdef-font-family-system-ui)

> This generic font family lets text render with the default user interface font (which, as with all [§ 2.1.5 Generic font families](https://drafts.csswg.org/css-fonts-4/#generic-font-families), may be a composite font) on the platform on which the UA is running. A cross-platform UA should use different fonts on its different supported platforms. The purpose of [system-ui](https://drafts.csswg.org/css-fonts-4/#valdef-font-family-system-ui) is to allow web content to integrate with the look and feel of the native OS. This is useful for hybrid, web or cross-platform applications.

> The precise font(s) used will depend on factors such as platform support, platform language, locale settings, user preferences, Unicode coverage and the content language. UA may optionally choose to use the text rendering engine provided by the platform when using [system-ui](https://drafts.csswg.org/css-fonts-4/#valdef-font-family-system-ui) to match the appearance of the default user interface more closely.

Nothing in that text says that the system font won't map to anything; although it may map to several somethings.

_(I realize you make additional points about remapping generics, but would like to understand your initial point before going further)_
